sympy/sets/sets.py

```diff
@@ -1577,6 +1583,12 @@ def as_relational(self, symbol):
 
         return And(A_rel, B_rel)
 
+    def _eval_evalf(self, prec):
+        try:
+            return self.func(*[arg._evalf(prec) for arg in self.args])
+        except (TypeError, ValueError, NotImplementedError):
+            raise TypeError("Not all sets are evalf-able")
```

Comment:
Is this not the same as the method inherited from the superclass?

Reply:
Oh I see the method isn't defined on `Set`.

Reply:
Maybe we should define it on `Set` and then only change it for things like `EmptySet` and `UniversalSet`.

Reply:
Does the same implementation not work for `EmptySet` which has empty args?

Reply:
It does. Thanks. I had no clue that the same implementation worked.